Jellyfin.Server.Implementations/AudioWaveform/AudioWaveformManager.cs

```diff
@@ -0,0 +1,160 @@
+            ?? throw new ResourceNotFoundException();
+
+        var saveInMediaFolder = BaseItem.LibraryManager.GetLibraryOptions(item).SaveLocalMetadata;
+        var saveFileName = Path.ChangeExtension(item.Path, "json");
+        string outputPathJson = saveInMediaFolder
+            ? Path.Combine(item.ContainingFolderPath, saveFileName)
+            : Path.Combine(item.GetInternalMetadataPath(), saveFileName);
+
+        if (!File.Exists(outputPathJson))
+        {
+            var tempDir = Path.Combine(_appPaths.TempDirectory, "audiowaveform_" + itemId);
+            Directory.CreateDirectory(tempDir);
+            string outputPathCsv = Path.Combine(tempDir, Path.ChangeExtension(saveFileName, "csv"));
+            int sampleRate = 0;
+            var mediaStreams = item.GetMediaStreams();
+            // error : mediaStreams[0].SampleRate can be null ? but why ? it is checked just before with .HasValue.
+            int fileSampleRate = (mediaStreams.Count > 0 && mediaStreams[0].SampleRate.HasValue) ? mediaStreams[0].SampleRate.Value : 44100;
+            // This value sets the length of a frame, the amount of measurements per second.
+            int samplesPerSecond = 2;
+            sampleRate = Math.Max(fileSampleRate / samplesPerSecond, 1);
+
+            // escape double quotes
+            string itemPathEscaped = item.Path.Replace("\"", "\\\"", StringComparison.Ordinal);
+            string outputPathCsvEscaped = outputPathCsv.Replace("\"", "\\\"", StringComparison.Ordinal);
+
+            var process = Process.Start("ffprobe", $"-v error -f lavfi -i \"amovie={itemPathEscaped},asetnsamples={sampleRate},astats=metadata=1:reset=1\" -show_entries frame_tags=lavfi.astats.Overall.RMS_peak -of csv=p=0 -o \"{outputPathCsvEscaped}\"");
+
+            int timeout = 10; // seconds, this is quite an arbitrary value
+            await process.WaitForExitAsync().WaitAsync(TimeSpan.FromSeconds(timeout)).ConfigureAwait(false);
+
+            var csvLines = await File.ReadAllLinesAsync(outputPathCsv).ConfigureAwait(false);
+            var samples = csvLines
+                    .Select(v =>
+                {
+                    if (string.Equals(v.Trim(), "-inf", StringComparison.Ordinal))
+                    {
+                        return 0.0;  // sometimes ffprobe returns "-inf" instead of a number
+                    }
+                    else
+                    {
+                        return double.TryParse(v, NumberStyles.Float, CultureInfo.InvariantCulture, out double parsedValue)
+                            ? Math.Pow(2, parsedValue / 6)
+                            : 0.0;
+                    }
+                })
+                .ToArray();
+
+            var data = new Dictionary<string, object>
+            {
+                { "samples", samples },
+                { "sampleRate", sampleRate }
+            };
+
+            var json = JsonSerializer.Serialize(data);
```

Comment:
```suggestion
            var json = JsonSerializer.Serialize(data, JsonDefaults.Options);
```
Always set the JsonSerializerOptions as this is where the typecache lives